captum/attr/_core/lime.py

```diff
@@ -411,8 +417,22 @@ def attribute(
             curr_model_inputs = []
             expanded_additional_args = None
             expanded_target = None
+            perturb_generator = None
+            if inspect.isgeneratorfunction(self.perturb_func):
+                perturb_generator = self.perturb_func(inputs, **kwargs)
+            batch_count = 0
             for _ in range(n_samples):
-                curr_sample = self.perturb_func(inputs, **kwargs)
+                if perturb_generator:
+                    try:
+                        curr_sample = next(perturb_generator)
+                    except StopIteration:
+                        warnings.warn(
+                            "Generator completed prior to given n_samples iterations!"
+                        )
+                        break
```

Comment:
Yes, I think this is needed since curr_sample wouldn't obtain a new sample for the remaining code in the loop. The alternative would be to raise an exception here rather than a warning, but it seems ok to just train with all samples returned from the generator, even if this is less than n_samples.